src/index/hgraph_index.cpp

```diff
@@ -79,7 +79,7 @@ HGraphIndex::Init() {
     mult_ = 1 / log(1.0 * static_cast<double>(this->bottom_graph_->MaximumDegree()));
 
     this->pool_ = std::make_shared<hnswlib::VisitedListPool>(
```

Comment:
be careful the `VisitedListPool` object and `allocator_` object destroy sequence
here can let VisitedListPool to hold shared_ptr ?

Reply:
I think you are right, I want remove the dependance of the hnswlib in hgraph, another pr #123 will introduce new visited pool.  now fix it on hnswlib's visited_pool

Reply:
OK